### ISSUE TITLE
VB -> C#: DataTable access by index wrong conversion

### ISSUE THREAD

**Issue by StealthyDeveloper** [icsharpcode/CodeConverter#720]
### Input code
```vbnet

Private _myTable As DataTable

CInt(_myTable(0)("MyColumnName"))
```

### Erroneous output
```C#

private DataTable _myTable;

Conversions.ToInteger(_myTable.ElementAtOrDefault(0)["MyColumnName"])
```
### Expected output
```C#

private DataTable _myTable; 

Conversions.ToInteger(_myTable.Rows[0]["MyColumnName"])
```

### Details
* Product in use: VS extension
* Version in use: 8.2.3.0
* Not seen a fix in older versions.
* If possible, could there be a fix in a few days? Otherwise I'd be willing to contribute, seeing as the conversion can't really wait any longer.

**Comment by GrahamTheCoder** [comment]
Sorry for the delay getting back to you. Hope the conversion went well if it went ahead!

When I try to compile your input code it issues an error for that one: [sharplab example](https://sharplab.io/#v2:DYLgbgRgNAJiDUAfAkgWwA4HsBOAXAzgAQDKAnvrgKaoB0AIgIa4MCwAUOwAoCuEwAlgGNCAYWAN8RZJO6UA7ACYADO0JrCnbPzBNKJABYNslGIQD6qUgBUGfPQEEijZjbur1m7bsIAPQgF5RZAA7XAAKC2tbYEowpQBKMIAiAFlSEUxgblRggDkGVEok+Pj2AFFg0zEJfCA)

Presumably DataTable is one of your own custom types Rows as the "default property", rather than  `System.Data.DataTable` that I'm trying to compile against?

**Comment by StealthyDeveloper** [comment]
No problem I've been busy as well and implemented a workaround for this in the source code. The exact DataTable type being used originates from System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089, .NET Framework v4.8. This code compiles for me in LINQPad5: [http://share.linqpad.net/79upfh.linq](url) which uses v.4.7.2. 

What's even more interesting is that hovering over the "indexer" in LPad also shows ElementAtOrDefault. Looking into System.Data.DataTable the default property is TableName not Rows oddly enough.

**Comment by GrahamTheCoder** [comment]
I've just checked in Visual Studio, and I can compile the same example there. A bit of googling showed that others have seen this oddity where it works sometimes and not others: https://forum.uipath.com/t/class-system-data-datatable-cannot-be-indexed-error/95620/10
https://forums.asp.net/t/1634320.aspx?class+system+data+datatable+cannot+be+indexed+because+it+has+no+default+property

**Comment by StealthyDeveloper** [comment]
Yes I think you're right. I think it's related to this one, but also for IListSource https://stackoverflow.com/questions/15864783/why-can-i-apply-an-indexer-to-an-icollection-in-vb-net-but-not-in-c-sharp.

**Comment by GrahamTheCoder** [comment]
Good find! In this case I'm wondering if it's essentially a chaining effect, where the data table has a default property, then that default property is ienumerable, allowing the indexer to work directly. Sounds like the sort of thing that might differ based on option strict

**Comment by GrahamTheCoder** [comment]
I've just noticed that DataTable has an `AsEnumerable` extension method in `DataTableExtensions`. I suspect this is one of those "duck-typed" methods that can be detected and used by the vb compiler.

Therefore I think the correct output might be:
```csharp
private DataTable _myTable; 

Conversions.ToInteger(_myTable.AsEnumerable().ElementAtOrDefault(0)["MyColumnName"])
```

**Comment by StealthyDeveloper** [comment]
Hi,

I had a look at the IL and the correct output should be like you said AsEnumerable().ElementAtOrDefault(0). Rows[index] is actually not the same since it throws on index > length rows. I set out a pr. I could'nt find an automatic way process this case using AsEnumerable() since all the related symbols I could find only mention ElementAtOrDefault(index), besides indirectly specifying when to add AsEnumerable() to the output.

**Comment by GrahamTheCoder** [comment]
Fixed in #744

**Comment by GrahamTheCoder** [comment]
By the way, if you write "Fixes #744" in an issue, it will auto-close the issue when merged

**Comment by StealthyDeveloper** [comment]
> By the way, if you write "Fixes #744" in an issue, it will auto-close the issue when merged

Thanks for this!

**Comment by GrahamTheCoder** [comment]
Ah whoops, I mean if you write "Fixes #744" in the **PR** description (or a commit message), obviously not in the issue like I said! 😄

**Comment by StealthyDeveloper** [comment]
> Ah whoops, I mean if you write "Fixes #744" in the **PR** description (or a commit message), obviously not in the issue like I said! 😄

I figured you meant that, but thanks for clarifying 👍
